src/utils/series-utils.js

```diff
@@ -108,44 +108,60 @@ function prepareData(data) {
  * @returns {Array} New array of children for the series.
  */
 export function getStackedData(children, attr) {
+  // It stores the last segment position added to each bar, separated by cluster.
+  const latestAttrPositions = {};
+
   return children.reduce((accumulator, series, seriesIndex) => {
     // Skip the children that are not series (e.g. don't have any data).
     if (!series) {
-      accumulator.result.push(null);
+      accumulator.push(null);
       return accumulator;
     }
 
     const {data, cluster = 'default'} = series.props;
     const preppedData = prepareData(data, attr);
 
     if (!attr || !preppedData || !preppedData.length) {
-      accumulator.result.push(preppedData);
+      accumulator.push(preppedData);
       return accumulator;
     }
 
     const attr0 = `${attr}0`;
+    const baseAttr = attr === 'y' ? 'x' : 'y';
+
+    accumulator.push(preppedData.map((d, dIndex) => {
+      if (!(cluster in latestAttrPositions)) {
```

Comment:
that's a pretty python-y way to say that, maybe better to say !latestAttrPositions[cluster]

Reply:
Thanks for the response. I just wanted to confirm that is not me misunderstanding something before putting in more effort in it. I will add some tests and fix some of the issues these days and will commit again.